iothub/service/src/Registry/DevicesClient.cs

```diff
@@ -262,6 +239,12 @@ public virtual async Task DeleteAsync(string deviceId, CancellationToken cancell
         /// To force the operation to execute regardless of ETag, set the device identity's ETag to "*" or
         /// use <see cref="DeleteAsync(string, CancellationToken)"/>.
         /// </param>
+        /// <param name="onlyIfUnchanged">
+        /// If false, this delete operation will be performed even if the provided device identity has
+        /// an out of date ETag. If true, the operation will throw a <see cref="PreconditionFailedException"/>
```

Comment:
Does the service return an error that we map to this exception?

Reply:
Yes, but the exception's shape is subject to change based on Bryce's work